prow/github/hmac.go

```diff
@@ -46,3 +64,39 @@ func PayloadSignature(payload []byte, key []byte) string {
 	sum := mac.Sum(nil)
 	return "sha1=" + hex.EncodeToString(sum)
 }
+
+// extractHmacs returns all *valid* HMAC tokens for given repository/organization. It iterates over full tree as necessary (repo->org->global)
+func extractHmacs(repo string, tokenGenerator func() []byte) [][]byte {
+	t := tokenGenerator()
+	repoToTokenMap := map[string]HmacsForRepo{}
+
+	err := yaml.Unmarshal(t, &repoToTokenMap)
```

Comment:
This will fail for all existing prows, which isn't something we can ship. You need to support the existing secret format (i.e. "just a string"), too.

Unfortunately, that probably means that you won't be able to do much meaningful error handling, because the old format was an arbitrary string...